cdap-data-fabric/src/test/java/co/cask/cdap/api/dataset/lib/SystemDatasetDefinitionTest.java

```diff
@@ -227,6 +229,39 @@ public void testPFSReconfigure() {
     } catch (IncompatibleUpdateException e) {
       // expected
     }
+
+    // test reconfiguring a PFS created before CDAP-13120, with no defaulted base path
+    // no base path should be set for the reconfigured dataset either
+    DatasetProperties oldProps = PartitionedFileSetProperties.builder()
+      .setPartitioning(Partitioning.builder().addStringField("s").build())
+      .add(PartitionedFileSetDefinition.NAME_AS_BASE_PATH_DEFAULT, "false")
+      .build();
+    DatasetSpecification oldSpec = pfsDef.configure("pfs", oldProps);
+    try {
```

Comment:
since this is a test, you could just add "throws IncompatibleUpdateException" to the method, and then you don't need to catch and wrap it here.